racun-service/src/main/java/rs/edu/raf/banka/racun/controller/UgovorController.java

```diff
@@ -0,0 +1,163 @@
+package rs.edu.raf.banka.racun.controller;
+
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+import rs.edu.raf.banka.racun.model.Ugovor;
+import rs.edu.raf.banka.racun.requests.*;
+import rs.edu.raf.banka.racun.service.impl.UgovorService;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/ugovor")
+public class UgovorController
+{
+    private final UgovorService ugovorService;
+
+    public UgovorController(UgovorService ugovorService) {
+        this.ugovorService = ugovorService;
+    }
+
+    @GetMapping(value = "/get/id/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> getUgovor(@RequestHeader("Authorization") String token, @PathVariable Long id) {
+        var ugovor = ugovorService.getById(id);
+        if(ugovor == null)
+            return ResponseEntity.badRequest().body("Ugovor not found");
+
+        return ResponseEntity.ok(ugovor);
+    }
+
+    @GetMapping(value = "/get/", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> getAll(@RequestHeader("Authorization") String token) {
+        var ugovori = ugovorService.getAll();
+        return ResponseEntity.ok(ugovori);
+    }
+
+    @GetMapping(value = "/get/finalized/", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> getAllFinalized(@RequestHeader("Authorization") String token) {
+        var ugovori = ugovorService.getAllFinalized();
+        return ResponseEntity.ok(ugovori);
+    }
+
+    @GetMapping(value = "/get/draft/", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> getAllDraft(@RequestHeader("Authorization") String token) {
+        var ugovori = ugovorService.getAllDraft();
+        return ResponseEntity.ok(ugovori);
+    }
+
+    @GetMapping(value = "/get/company/{kompanijaId}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> getAllKompanija(@RequestHeader("Authorization") String token, @PathVariable Long kompanijaId) {
+        try {
+            List<Ugovor> ugovori = ugovorService.getAllByCompany(kompanijaId);
+            return ResponseEntity.ok(ugovori);
+
+        } catch (Exception ex) {
+            return ResponseEntity.badRequest().body(ex.getMessage());
+        }
```

Comment:
Generalno se ovo ne radi, već se pravi ExceptionHandler koji će da napakuje poruku u neku strukturu koja može da se parsira. Do sada nismo radili ovo, tako da neka ovako kako je, ali čisto napomena pošto ćemo u nekom trenutku ovo da izmenimo.